src/qibo/tensorflow/cgates.py

```diff
@@ -797,5 +802,100 @@ def __init__(self, q: int, px: float = 0, py: float = 0, pz: float = 0,
 
     @staticmethod
     def _invert(gate):
-        """For Pauli gates we can use same gate for state inversion for efficiency."""
+        # for Pauli gates we can use same gate as inverse for efficiency
         return gate
+
+
+class ResetChannel(UnitaryChannel, base_gates.ResetChannel):
```

Comment:
I have a couple of questions here:
- I see that the inheritance from MatrixGate is used in H, however not for X, Y, Z, is that correct?
- this class inherits from UnitaryChannel, but you don't initialize its constructor because you are planning to use just the `_density_matrix_call`, correct? (and not the _prepare which needs the KrausChannel)

Reply:
> * I see that the inheritance from MatrixGate is used in H, however not for X, Y, Z, is that correct?

Yes, `MatrixGate` is essentially the "abstract" class for gates that use the `op.apply_gate` or `op.apply_two_qubit_gate` custom operators which take a gate `matrix` argument. X, Y and Z do not inherit this because they are not applied via matrix multiplication but via the `op.apply_x`, `op.apply_y`, `op.apply_z` which have the "matrices" of each gate hard-coded in the C++ code.

> * this class inherits from UnitaryChannel, but you don't initialize its constructor because you are planning to use just the _density_matrix_call, correct? (and not the _prepare which needs the KrausChannel)

Inheritance is a bit complicated to follow with channels, but I am not sure if there is any way to simplify things. For example, you may notice that I never call the constructor of the parent channel in the backend modules (inside the `tensorflow` folder) because the parent constructor is called from the base constructor.

Regarding the `ResetChannel`, it inherits both the `_prepare` and `_density_matrix_call` methods from the `UnitaryChannel`. Also, channels that inherit the `UnitaryChannel` are the ones that can be used with state vectors / sampling. The others can only be used with density matrices.